### PR TITLE
Feat/View Routes

### PR DESCRIPTION
**Pull request by qwoprocks**
Summary:
- Fix bug where "Upload wall image" didn't work
- Add default routing for random paths etc 'routermaker.rocks/aijmeine' to redirect to '/home'
- Update outdated packages
- Add route cards when selecting a gym
- Allow user to hide and unhide a map
- Add view route page, complete with comments, report buttons, grading, deleting comments.

### CODE REVIEW

**Comment by yarkhinephyo** [comment]
`Upload Wall Image` button has stopped working.

File: ionic_user_interface/src/views/ViewRoute.vue [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+  },
+});
+</script>
+
+<style scoped>
+#container {
+  position: absolute;
+  left: 0;
+  right: 0;
+  max-width: 1000px;
```
Comment (yarkhinephyo) [inline ionic_user_interface/src/views/ViewRoute.vue:497]: Too big for desktop.. 800px?

File: ionic_user_interface/src/views/ViewRoute.vue [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+                          text: 'Close',
+                          role: 'cancel',
+                        },
+                      ],
+                    })
+                    .then((toast) => {
+                      toast.present();
+                    });
+                })
+                .catch((error) => {
```
Comment (yarkhinephyo) [inline ionic_user_interface/src/views/ViewRoute.vue:444]: On reporting multiple times,

HTTP 400
```
{
    "Message": "Duplicate route report"
}
```

File: ionic_user_interface/src/views/ViewRoute.vue (hunk 1 above)
Comment (qwoprocks) [reply ionic_user_interface/src/views/ViewRoute.vue]: Changed to 900px, 800px is too small

File: ionic_user_interface/src/views/ViewRoute.vue (hunk 2 above)
Comment (qwoprocks) [reply ionic_user_interface/src/views/ViewRoute.vue:453]: I already disabled the button if `hasReport == true`

File: ionic_user_interface/src/views/ViewRoute.vue [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+            :disabled="false"
+            :changeHandler="gradeChangeHandler"
+          ></GradeSlider>
+        </div>
+        <br />
+        <ion-row
+          v-if="!hasAlreadyCommented"
+          class="ion-align-items-start ion-justify-content-start"
+        >
+          <ion-textarea
```
Comment (yarkhinephyo) [inline ionic_user_interface/src/views/ViewRoute.vue:50]: Can add a small border radius

File: ionic_user_interface/src/views/ViewRoute.vue [hunk 4]
```diff
@@ -0,0 +1,591 @@
...
+        })
+        .catch((error) => {
+          console.error(error);
+          router.back();
+        });
+    }, 1000);
+
+    updateRouteDetails();
+
+    const postCommentHandler = throttle(() => {
```
Comment (yarkhinephyo) [inline ionic_user_interface/src/views/ViewRoute.vue:229]: Add ascii validation

File: ionic_user_interface/src/components/GradeSlider.vue [hunk 5]
```diff
@@ -0,0 +1,109 @@
+<template>
+  <div class="range">
+    <ion-range
+      class="slider"
+      type="range"
+      min="0"
+      max="15"
```
Comment (yarkhinephyo) [inline ionic_user_interface/src/components/GradeSlider.vue:7]: maximum grade in backend is 14. following moonboard app

File: ionic_user_interface/src/views/ViewRoute.vue [hunk 6]
```diff
@@ -0,0 +1,591 @@
...
+      commentText.value = commentText.value.trim();
+      if (commentText.value.length === 0) {
+        console.log('Comment cannot be empty');
```
Comment (yarkhinephyo) [inline ionic_user_interface/src/views/ViewRoute.vue:232]: I think you meant to use `Msgbox`

File: ionic_user_interface/src/components/RouteList.vue [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        : {};
+      axios
+        .get(process.env.VUE_APP_ROUTE_ENDPOINT_URL + '/route/all', {
+          headers,
+          params: {
+            gymLocation,
+          },
+        })
+        .then((response) => {
+          console.log(response);
```
Comment (yarkhinephyo) [inline ionic_user_interface/src/components/RouteList.vue:84]: We can remove this

File: ionic_user_interface/src/views/ViewRoute.vue [hunk 8]
```diff
@@ -0,0 +1,591 @@
...
+        .then((response) => {
+          if (response.data.Message === 'Get route details success') {
+            routeDetails.value = response.data.Item;
+            hasAlreadyCommented.value =
+              routeDetails.value.comments?.some(
+                (comment) => comment.username == myUsername.value,
+              ) ?? false;
+            hasReported.value = routeDetails.value.hasReported ?? false;
+            hasLoaded.value = true;
+            console.log(response.data.Item);
```
Comment (yarkhinephyo) [inline ionic_user_interface/src/views/ViewRoute.vue:218]: We can remove this too

File: ionic_user_interface/src/views/ViewRoute.vue [hunk 9]
```diff
@@ -0,0 +1,591 @@
...
+            grade,
+          },
+          {
+            headers: {
+              Authorization: `Bearer ${getAccessToken().value}`,
+            },
+          },
+        )
+        .then((response) => {
+          console.log(response);
```
Comment (yarkhinephyo) [inline ionic_user_interface/src/views/ViewRoute.vue:498]: Can remove this too

**Review by yarkhinephyo — APPROVED** [review]
Functions all working. User interactions like posting comment, deleting comment can put loading spinner but up to you

**Pull request merged**